pkg/graveler/committed/merge_test.go

```diff
@@ -1082,59 +1107,489 @@ func Test_merge(t *testing.T) {
+							key:      "a",
+							identity: "a1",
+						},
+						{
+							action:   actionTypeWriteRecord,
+							key:      "c",
+							identity: "c",
+						},
+						{
+							action:   actionTypeWriteRecord,
+							key:      "d",
+							identity: "d",
+						},
+					},
+					expectedErr: nil,
+				},
+				{
+					mergeStrategies:     []graveler.MergeStrategy{graveler.MergeStrategyTheirs},
+					conflictExpectedIdx: nil,
+					expectedActions: []writeAction{
+						{
+							action:   actionTypeWriteRecord,
+							key:      "c",
+							identity: "c",
+						},
+						{
+							action:   actionTypeWriteRecord,
+							key:      "d",
+							identity: "d",
+						},
+					},
+					expectedErr: nil,
+				},
+			},
+		},
+		"source change and dest delete same entry": {
+			baseRange: []testRange{{
+				rng:     committed.Range{ID: "base:a-c", MinKey: committed.Key("a"), MaxKey: committed.Key("c"), Count: 3, EstimatedSize: 333},
+				records: []testValueRecord{{key: "a", identity: "a"}, {key: "b", identity: "b"}, {key: "c", identity: "c"}},
+			}},
+			sourceRange: []testRange{{
+				rng:     committed.Range{ID: "source:a-c", MinKey: committed.Key("a"), MaxKey: committed.Key("c"), Count: 3, EstimatedSize: 123},
+				records: []testValueRecord{{key: "a", identity: "a"}, {key: "b", identity: "b1"}, {key: "c", identity: "c"}},
+			}},
+			destRange: []testRange{{
+				rng:     committed.Range{ID: "dest:a-c", MinKey: committed.Key("a"), MaxKey: committed.Key("c"), Count: 2, EstimatedSize: 321},
+				records: []testValueRecord{{key: "a", identity: "a"}, {key: "c", identity: "c"}},
+			}},
+			expectedResult: []testRunResult{
+				{
+					mergeStrategies:     []graveler.MergeStrategy{graveler.MergeStrategyNone},
+					conflictExpectedIdx: nil,
+					expectedActions: []writeAction{
+						{
+							action:   actionTypeWriteRecord,
+							key:      "a",
+							identity: "a",
+						},
+					},
+					expectedErr: graveler.ErrConflictFound,
+				},
+				{
+					mergeStrategies:     []graveler.MergeStrategy{graveler.MergeStrategyOurs},
+					conflictExpectedIdx: nil,
+					expectedActions: []writeAction{
+						{
+							action:   actionTypeWriteRecord,
+							key:      "a",
+							identity: "a",
+						},
+						{
+							action:   actionTypeWriteRecord,
+							key:      "c",
+							identity: "c",
+						},
+					},
+					expectedErr: nil,
+				},
+				{
+					mergeStrategies:     []graveler.MergeStrategy{graveler.MergeStrategyTheirs},
+					conflictExpectedIdx: nil,
+					expectedActions: []writeAction{
+						{
+							action:   actionTypeWriteRecord,
+							key:      "a",
+							identity: "a",
+						},
+						{
+							action:   actionTypeWriteRecord,
+							key:      "b",
+							identity: "b1",
+						},
+						{
+							action:   actionTypeWriteRecord,
+							key:      "c",
+							identity: "c",
+						},
+					},
+					expectedErr: nil,
+				},
+			},
+		},
+
+		"source delete and dest change same entry": {
```

Comment:
It doesn't test this case. It's testing source deleted dest changes. Change it to source b->b1 and dest b->b2

Reply:
Not sure I understand why it's not the case.
Eitherways, added another test, with the b->b1 and b->b2 conflict you suggested